### PR TITLE
Docs: Add Value to Default Label Provider Docs

### PR DESCRIPTION
**Pull request by rlingineni**
Adding TagData Remove label to docs table

https://gestalt.pinterest.systems/web/utilities/defaultlabelprovider#Default-strings-provided

### CODE REVIEW

**Comment by AlbertCarreras** [comment]
https://github.com/pinterest/gestalt/blob/c5ee1613bb3241203690d20f19b2338693f483e1/packages/gestalt/dist/index.d.ts#L345 


Can yo add it here as well?

**Review by AlbertCarreras — CHANGES_REQUESTED** [review]
add type to TS file

**Comment by rlingineni** [comment]
@AlbertCarreras - added, and also updated readme to remind adding ts defs

File: scripts/templates/README.md [hunk 1]
```diff
@@ -103,3 +103,8 @@ yarn playwright:update-visual-test playwright/visual-test/ComponentName --update
 
 `.github/pr-title-checker-config.json`
 Add your new component name to the regex (alphabetize please!). If you're renaming, replacing, or removing a component, — _and the component also doesn't exist on any other platforms_ — remove that component name from the regex.
+
```
Comment (AlbertCarreras) [inline scripts/templates/README.md:106]: See line 76 @rlingineni

File: packages/gestalt/src/TagData.js [hunk 2]
```diff
@@ -44,11 +44,7 @@ export type TileChangeHandler = ({|
 |}) => void;
 
 export type OnRemoveHandler = ({|
-  event:
-    | SyntheticMouseEvent<HTMLDivElement>
-    | SyntheticKeyboardEvent<HTMLDivElement>
-    | SyntheticMouseEvent<HTMLAnchorElement>
-    | SyntheticKeyboardEvent<HTMLAnchorElement>,
+  event: SyntheticMouseEvent<HTMLButtonElement>,
```
Comment (AlbertCarreras) [inline packages/gestalt/src/TagData.js:47]: Why this change?

File: packages/gestalt/dist/index.d.ts [hunk 3]
```diff
@@ -1864,6 +1867,29 @@ interface TagProps {
   type?: 'default' | 'error' | 'warning' | undefined;
 }
 
+interface TagDataProps {
+  accessibilityRemoveIconLabel?: string | undefined;
+  baseColor?: 'primary' | 'secondary'
+  color?: DataVisualizationColors,
+  disabled?: boolean,
+  id?: string,
+  onTap?: | AbstractEventHandler<
```
Comment (AlbertCarreras) [inline packages/gestalt/dist/index.d.ts:1876]: have forced prettier here?

the initial union is needed? `"|" Abs`

File: packages/gestalt/src/TagData.js (hunk 2 above)
Comment (AlbertCarreras) [reply packages/gestalt/src/TagData.js:47]: what if it's click from the keyboard?

**Review by AlbertCarreras — APPROVED** [review]
R+ but look at the comments

File: packages/gestalt/src/TagData.js (hunk 2 above)
Comment (rlingineni) [reply packages/gestalt/src/TagData.js:47]: Same handler/implementation as`Tag`. It's an HTML button `onClick` handler, it doesn't give the keyboard events back

File: scripts/templates/README.md (hunk 1 above)
Comment (rlingineni) [reply scripts/templates/README.md:106]: Oh lol, I just can't read
